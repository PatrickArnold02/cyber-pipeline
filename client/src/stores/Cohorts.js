// Imports
import { defineStore } from 'pinia'
import Logger from 'js-logger'

// Services
import api from '@/services/api'

export const useCohortsStore = defineStore('cohorts', {
  state: () => {
    return {
      cohorts: [], // list of cohorts
      dashStatus: [], // status dashboard
      pdStatus: [], // pd status dashboard
      certStatus: [], // cert status dashboard
      msStatus: [] // ms status dashboard
    }
  },
  getters: {
    /**
     * Getter for an individual item
     *
     * @param {State} state
     * @returns a function to find an item based on a given id
     */
    getCohort: (state) => {
      return (id) => state.cohorts.find((cohort) => cohort.id === id)
    }
  },
  actions: {
    /**
     * Hydrate the store by querying the API for data
     */
    async hydrate() {
      Logger.info('cohorts:hydrate')
      await api.get('/api/v1/cohorts').then((response) => {
        this.cohorts = response.data
      })
      await api.get('/api/v1/dashboard/cohort/status').then((response) => {
        this.dashStatus = response.data
      })
      await api.get('/api/v1/dashboard/cohort/pdstatus').then((response) => {
        this.pdStatus = response.data
      })
      await api.get('/api/v1/dashboard/cohort/certstatus').then((response) => {
        this.certStatus = response.data
      })
      await api.get('/api/v1/dashboard/cohort/msstatus').then((response) => {
        this.msStatus = response.data
      })
    },

    /**
     * Update an item via the API
     *
     * @param {Cohort} cohort
     */
    async update(cohort) {
      await api
        .post('/api/v1/cohorts/' + cohort.id, {
          cohort: cohort
        })
        .then(async () => {
          await this.hydrate()
        })
    },

    /**
     * Create a new item via the API
     *
     * @param {Cohort} cohort
     */
    async new(cohort) {
      await api
        .put('/api/v1/cohorts', {
          cohort: cohort
        })
        .then(async () => {
          await this.hydrate()
        })
    },
    /**
     * Delete an item with the given ID via the API
     *
     * @param {Integer} id
     */
    async delete(id) {
      await api.delete('/api/v1/cohorts/' + id).then(async () => {
        await this.hydrate()
      })
    }
  }
})
