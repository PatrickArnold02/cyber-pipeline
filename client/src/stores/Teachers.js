// Imports
import { defineStore } from 'pinia'
import Logger from 'js-logger'

// Services
import api from '@/services/api'

export const useTeachersStore = defineStore('teachers', {
  state: () => {
    return {
      teachers: [] // list of teachers
    }
  },
  getters: {
    /**
     * Getter for an individual item
     *
     * @param {State} state
     * @returns a function to find an item based on a given id
     */
    getTeacher: (state) => {
      return (id) => state.teachers.find((teacher) => teacher.id === id)
    }
  },
  actions: {
    /**
     * Hydrate the store by querying the API for data
     */
    async hydrate() {
      Logger.info('teachers:hydrate')
      await api.get('/api/v1/teachers').then((response) => {
        this.teachers = response.data
        this.teachers.forEach((teacher) => {
          teacher.districts.map((district) => {
            district.primary = district.primary === 1
          })
          teacher.all_districts = teacher.districts.map((district) => district.usdName).join(', ')
        })
      })
    },

    /**
     * Update an item via the API
     *
     * @param {Teacher} teacher
     */
    async update(teacher) {
      teacher.districts.map((district) => {
        district.primary = district.primary ? 1 : 0
      })
      await api
        .post('/api/v1/teachers/' + teacher.id, {
          teacher: teacher
        })
        .then(async () => {
          await this.hydrate()
        })
    },

    /**
     * Create a new item via the API
     *
     * @param {Teacher} teacher
     */
    async new(teacher) {
      teacher.districts.map((district) => {
        district.primary = district.primary ? 1 : 0
      })
      await api
        .put('/api/v1/teachers', {
          teacher: teacher
        })
        .then(async () => {
          await this.hydrate()
        })
    },

    /**
     * Delete an item with the given ID via the API
     *
     * @param {Integer} id
     */
    async delete(id) {
      await api.delete('/api/v1/teachers/' + id).then(async () => {
        await this.hydrate()
      })
    }
  }
})
