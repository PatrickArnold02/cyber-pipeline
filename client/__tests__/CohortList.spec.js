import { mount } from '@vue/test-utils'
import { describe, it, expect, beforeEach, vi, afterEach } from 'vitest'
import CohortList from '../src/components/cohort/CohortList.vue'
import { useCohortsStore } from '../src/stores/Cohorts.js'
import { useTeachersStore } from '../src/stores/Teachers.js'
import Panel from 'primevue/panel'
import DataTable from 'primevue/datatable'
import Button from 'primevue/button'
import Popover from 'primevue/popover'
import Dialog from 'primevue/dialog'
import ConfirmDialog from 'primevue/confirmdialog'
import { ref } from 'vue'
import { nextTick } from 'vue'

// Mocking the stores
vi.mock('../src/stores/Cohorts')
vi.mock('../src/stores/Teachers')

describe('CohortList', () => {
  let wrapper;
  let cohortsStore;
  let teachersStore;

  beforeEach(() => {
    cohortsStore = {
      hydrate: vi.fn(),
      delete: vi.fn(),
      update: vi.fn(),
      new: vi.fn(),
      cohorts: ref([{ id: 1, name: 'Test Cohort' }]) // Added a test cohort
    };

    teachersStore = {
      hydrate: vi.fn(),
      teachers: ref([{ id: 1, name: 'Test Teacher' }]) // Added a test teacher
    };

    useCohortsStore.mockReturnValue(cohortsStore);
    useTeachersStore.mockReturnValue(teachersStore);

    // Mount the component with props and reactive properties
    wrapper = mount(CohortList);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('renders correctly', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.findComponent(Panel).exists()).toBe(true);
    expect(wrapper.findComponent(Panel).text()).toContain('Manage Cohorts');

    expect(wrapper.findComponent(Button).exists()).toBe(true);
    expect(wrapper.findComponent(Button).text()).toContain('New');
    expect(wrapper.findComponent(Button).props().icon).toBe('pi pi-plus');
  });

  it('calls the editCohort method', async () => {
    const cohort = { id: 1, name: 'Test Cohort' };
    const spy = vi.spyOn(wrapper.vm, 'editCohort');
    await wrapper.vm.editCohort(cohort);

    expect(spy).toHaveBeenCalledWith(cohort);
    expect(wrapper.vm.cohortDialog).toBe(true);
    expect(wrapper.vm.cohortDialogHeader).toBe('Edit Cohort');
    expect(wrapper.vm.cohort).toEqual(cohort);
  });

  it('calls the newCohort method', async () => {
    const spy = vi.spyOn(wrapper.vm, 'newCohort');
    await wrapper.vm.newCohort();

    expect(spy).toHaveBeenCalled();
    expect(wrapper.vm.cohortDialog).toBe(true);
    expect(wrapper.vm.cohortDialogHeader).toBe('New Cohort');
  });

  it('calls the deleteCohort method', async () => {
    const cohort = { id: 1, name: 'Test Cohort' };
    const spy = vi.spyOn(wrapper.vm, 'deleteCohort');
    const confirmSpy = vi.spyOn(wrapper.vm.$confirm, 'require');

    await wrapper.vm.deleteCohort(cohort);

    expect(spy).toHaveBeenCalledWith(cohort);
    expect(confirmSpy).toHaveBeenCalledWith({
      message: 'Are you sure you want to delete Test Cohort?',
      header: 'Danger Zone',
      icon: 'pi pi-exclamation-triangle',
      rejectLabel: 'Cancel',
      acceptLabel: 'Delete',
      rejectClass: 'p-button-secondary p-button-outlined',
      acceptClass: 'p-button-danger',
      accept: expect.any(Function),
      reject: expect.any(Function)
    });
  });

  it('calls the save method', async () => {
    const cohort = { id: 2, name: 'New Cohort', teachers: [] }; // Providing teachers
    wrapper.vm.cohort = cohort; 

    const spy = vi.spyOn(wrapper.vm, 'save');
    await wrapper.vm.save(cohort);

    expect(spy).toHaveBeenCalled();
    expect(wrapper.vm.cohortDialog).toBe(false);
  });

  it('exports the datatable to CSV', () => {
    wrapper.vm.dt = { exportCSV: vi.fn() };
    wrapper.vm.exportCSV();
    expect(wrapper.vm.dt.exportCSV).toHaveBeenCalled();
  });

  it('renders the DataTable', () => {
    expect(wrapper.findComponent(DataTable).exists()).toBe(true);
    expect(wrapper.findComponent(DataTable).props().value).toBe(wrapper.vm.cohorts);
  });

  it('renders the Dialog, ConfirmDialog, and Popover', () => {
    expect(wrapper.findComponent(Dialog).exists()).toBe(true);
    expect(wrapper.findComponent(ConfirmDialog).exists()).toBe(true);
    expect(wrapper.findComponent(Popover).exists()).toBe(true);
  });
});
