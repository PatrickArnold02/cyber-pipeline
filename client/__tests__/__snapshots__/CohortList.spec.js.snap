// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CohortList > matches the snapshot 1`] = `
"<!-- Location for confirmation dialog to be inserted -->
<!--teleport start-->
<!--teleport end-->
<!-- Main datatable for items -->
<div data-v-2fbb3249="" class="p-panel p-component" data-pc-name="panel" pc97="" data-pc-section="root">
  <div class="p-panel-header" data-pc-section="header"><span id="pv_id_27_header" class="p-panel-title" data-pc-section="title">Manage Cohorts</span>
    <div class="p-panel-header-actions" data-pc-section="headeractions">
      <!---->
    </div>
  </div>
  <transition-stub name="p-toggleable-content" appear="false" persisted="false" css="true">
    <div id="pv_id_27_content" class="p-panel-content-container" role="region" aria-labelledby="pv_id_27_header" data-pc-section="contentcontainer">
      <div class="p-panel-content" data-pc-section="content">
        <div data-v-2fbb3249="" class="p-datatable p-component p-datatable-striped" data-scrollselectors=".p-datatable-wrapper" data-pc-name="datatable" pc98="" data-pc-section="root">
          <!---->
          <!---->
          <!---->
          <div class="p-datatable-header" data-pc-section="header">
            <div data-v-2fbb3249="" class="p-toolbar p-component mb-2" role="toolbar" style="border: none none;" data-pc-name="toolbar" pc101="" data-pc-section="root">
              <div class="p-toolbar-start" data-pc-section="start"><button data-v-2fbb3249="" class="p-button p-component p-button-success mr-2" type="button" aria-label="New" data-pc-name="button" data-p-disabled="false" data-p-severity="success" pc102="" data-pc-section="root"><span class="p-button-icon p-button-icon-left pi pi-plus" data-pc-section="icon"></span><span class="p-button-label" data-pc-section="label">New</span>
                  <!---->
                </button><button data-v-2fbb3249="" class="p-button p-component p-button-help" type="button" aria-label="Export" data-pc-name="button" data-p-disabled="false" data-p-severity="help" pc103="" data-pc-section="root"><span class="p-button-icon p-button-icon-left pi pi-upload" data-pc-section="icon"></span><span class="p-button-label" data-pc-section="label">Export</span>
                  <!---->
                </button></div>
              <div class="p-toolbar-center" data-pc-section="center"></div>
              <div class="p-toolbar-end" data-pc-section="end">
                <div data-v-2fbb3249="" class="p-iconfield" iconposition="left" data-pc-name="iconfield" pc104="" data-pc-section="root"><span data-v-2fbb3249="" class="p-inputicon" data-pc-name="inputicon" pc105="" data-pc-section="root"><i data-v-2fbb3249="" class="pi pi-search"></i></span><input data-v-2fbb3249="" type="text" class="p-inputtext p-component" placeholder="Keyword Search" data-pc-name="inputtext" pc106="" data-pc-section="root" value=""></div>
              </div>
            </div>
          </div>
          <nav class="p-datatable-paginator-top" data-pc-section="paginatorcontainer">
            <div class="p-paginator p-component" data-pc-name="pcpaginator" data-pc-extend="paginator" pc107="" data-pc-section="root">
              <!---->
              <div class="p-paginator-content" data-pc-section="content"><button class="p-paginator-first p-disabled" type="button" data-pc-section="first" data-pc-group-section="pagebutton" aria-label="First Page" disabled=""><svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg" class="p-icon p-paginator-first-icon" aria-hidden="true" data-pc-section="firsticon">
                    <path fill-rule="evenodd" clip-rule="evenodd" d="M5.71602 11.164C5.80782 11.2021 5.9063 11.2215 6.00569 11.221C6.20216 11.2301 6.39427 11.1612 6.54025 11.0294C6.68191 10.8875 6.76148 10.6953 6.76148 10.4948C6.76148 10.2943 6.68191 10.1021 6.54025 9.96024L3.51441 6.9344L6.54025 3.90855C6.624 3.76126 6.65587 3.59011 6.63076 3.42254C6.60564 3.25498 6.525 3.10069 6.40175 2.98442C6.2785 2.86815 6.11978 2.79662 5.95104 2.7813C5.78229 2.76598 5.61329 2.80776 5.47112 2.89994L1.97123 6.39983C1.82957 6.54167 1.75 6.73393 1.75 6.9344C1.75 7.13486 1.82957 7.32712 1.97123 7.46896L5.47112 10.9991C5.54096 11.0698 5.62422 11.1259 5.71602 11.164ZM11.0488 10.9689C11.1775 11.1156 11.3585 11.2061 11.5531 11.221C11.7477 11.2061 11.9288 11.1156 12.0574 10.9689C12.1815 10.8302 12.25 10.6506 12.25 10.4645C12.25 10.2785 12.1815 10.0989 12.0574 9.96024L9.03158 6.93439L12.0574 3.90855C12.1248 3.76739 12.1468 3.60881 12.1204 3.45463C12.0939 3.30045 12.0203 3.15826 11.9097 3.04765C11.7991 2.93703 11.6569 2.86343 11.5027 2.83698C11.3486 2.81053 11.19 2.83252 11.0488 2.89994L7.51865 6.36957C7.37699 6.51141 7.29742 6.70367 7.29742 6.90414C7.29742 7.1046 7.37699 7.29686 7.51865 7.4387L11.0488 10.9689Z" fill="currentColor"></path>
                  </svg></button><button class="p-paginator-prev p-disabled" type="button" data-pc-section="prev" data-pc-group-section="pagebutton" aria-label="Previous Page" disabled=""><svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg" class="p-icon p-paginator-prev-icon" aria-hidden="true" data-pc-section="previcon">
                    <path d="M8.75 11.185C8.65146 11.1854 8.55381 11.1662 8.4628 11.1284C8.37179 11.0906 8.28924 11.0351 8.22 10.965L4.72 7.46496C4.57955 7.32433 4.50066 7.13371 4.50066 6.93496C4.50066 6.73621 4.57955 6.54558 4.72 6.40496L8.22 2.93496C8.36095 2.84357 8.52851 2.80215 8.69582 2.81733C8.86312 2.83252 9.02048 2.90344 9.14268 3.01872C9.26487 3.134 9.34483 3.28696 9.36973 3.4531C9.39463 3.61924 9.36303 3.78892 9.28 3.93496L6.28 6.93496L9.28 9.93496C9.42045 10.0756 9.49934 10.2662 9.49934 10.465C9.49934 10.6637 9.42045 10.8543 9.28 10.995C9.13526 11.1257 8.9448 11.1939 8.75 11.185Z" fill="currentColor"></path>
                  </svg></button><span class="p-paginator-pages" data-pc-section="pages"><button class="p-paginator-page p-paginator-page-selected" type="button" aria-label="Page 1" aria-current="page" data-pc-section="page" data-p-active="true">1</button></span><button class="p-paginator-next p-disabled" type="button" data-pc-section="next" data-pc-group-section="pagebutton" aria-label="Next Page" disabled=""><svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg" class="p-icon p-paginator-next-icon" aria-hidden="true" data-pc-section="nexticon">
                    <path d="M5.25 11.1728C5.14929 11.1694 5.05033 11.1455 4.9592 11.1025C4.86806 11.0595 4.78666 10.9984 4.72 10.9228C4.57955 10.7822 4.50066 10.5916 4.50066 10.3928C4.50066 10.1941 4.57955 10.0035 4.72 9.86283L7.72 6.86283L4.72 3.86283C4.66067 3.71882 4.64765 3.55991 4.68275 3.40816C4.71785 3.25642 4.79932 3.11936 4.91585 3.01602C5.03238 2.91268 5.17819 2.84819 5.33305 2.83149C5.4879 2.81479 5.64411 2.84671 5.78 2.92283L9.28 6.42283C9.42045 6.56346 9.49934 6.75408 9.49934 6.95283C9.49934 7.15158 9.42045 7.34221 9.28 7.48283L5.78 10.9228C5.71333 10.9984 5.63193 11.0595 5.5408 11.1025C5.44966 11.1455 5.35071 11.1694 5.25 11.1728Z" fill="currentColor"></path>
                  </svg></button><button class="p-paginator-last p-disabled" type="button" data-pc-section="last" data-pc-group-section="pagebutton" aria-label="Last Page" disabled=""><svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg" class="p-icon p-paginator-last-icon" aria-hidden="true" data-pc-section="lasticon">
                    <path fill-rule="evenodd" clip-rule="evenodd" d="M7.68757 11.1451C7.7791 11.1831 7.8773 11.2024 7.9764 11.2019C8.07769 11.1985 8.17721 11.1745 8.26886 11.1312C8.36052 11.088 8.44238 11.0265 8.50943 10.9505L12.0294 7.49085C12.1707 7.34942 12.25 7.15771 12.25 6.95782C12.25 6.75794 12.1707 6.56622 12.0294 6.42479L8.50943 2.90479C8.37014 2.82159 8.20774 2.78551 8.04633 2.80192C7.88491 2.81833 7.73309 2.88635 7.6134 2.99588C7.4937 3.10541 7.41252 3.25061 7.38189 3.40994C7.35126 3.56927 7.37282 3.73423 7.44337 3.88033L10.4605 6.89748L7.44337 9.91463C7.30212 10.0561 7.22278 10.2478 7.22278 10.4477C7.22278 10.6475 7.30212 10.8393 7.44337 10.9807C7.51301 11.0512 7.59603 11.1071 7.68757 11.1451ZM1.94207 10.9505C2.07037 11.0968 2.25089 11.1871 2.44493 11.2019C2.63898 11.1871 2.81949 11.0968 2.94779 10.9505L6.46779 7.49085C6.60905 7.34942 6.68839 7.15771 6.68839 6.95782C6.68839 6.75793 6.60905 6.56622 6.46779 6.42479L2.94779 2.90479C2.80704 2.83757 2.6489 2.81563 2.49517 2.84201C2.34143 2.86839 2.19965 2.94178 2.08936 3.05207C1.97906 3.16237 1.90567 3.30415 1.8793 3.45788C1.85292 3.61162 1.87485 3.76975 1.94207 3.9105L4.95922 6.92765L1.94207 9.9448C1.81838 10.0831 1.75 10.2621 1.75 10.4477C1.75 10.6332 1.81838 10.8122 1.94207 10.9505Z" fill="currentColor"></path>
                  </svg></button>
                <div class="p-select p-component p-inputwrapper p-inputwrapper-filled p-paginator-rpp-dropdown" data-pc-group-section="pagedropdown" data-pc-name="pcrowperpagedropdown" data-pc-extend="select" pc118="" data-pc-section="root" id="pv_id_26"><span class="p-select-label" tabindex="0" role="combobox" aria-label="Rows per page" aria-haspopup="listbox" aria-expanded="false" aria-controls="pv_id_26_list" aria-disabled="false" data-pc-section="label">5</span>
                  <!---->
                  <div class="p-select-dropdown" data-pc-section="dropdown"><svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg" class="p-icon p-select-dropdown-icon" aria-hidden="true" data-pc-section="dropdownicon">
                      <path d="M7.01744 10.398C6.91269 10.3985 6.8089 10.378 6.71215 10.3379C6.61541 10.2977 6.52766 10.2386 6.45405 10.1641L1.13907 4.84913C1.03306 4.69404 0.985221 4.5065 1.00399 4.31958C1.02276 4.13266 1.10693 3.95838 1.24166 3.82747C1.37639 3.69655 1.55301 3.61742 1.74039 3.60402C1.92777 3.59062 2.11386 3.64382 2.26584 3.75424L7.01744 8.47394L11.769 3.75424C11.9189 3.65709 12.097 3.61306 12.2748 3.62921C12.4527 3.64535 12.6199 3.72073 12.7498 3.84328C12.8797 3.96582 12.9647 4.12842 12.9912 4.30502C13.0177 4.48162 12.9841 4.662 12.8958 4.81724L7.58083 10.1322C7.50996 10.2125 7.42344 10.2775 7.32656 10.3232C7.22968 10.3689 7.12449 10.3944 7.01744 10.398Z" fill="currentColor"></path>
                    </svg></div>
                  <!--teleport start-->
                  <!--teleport end-->
                </div>
              </div>
              <!---->
            </div>
          </nav>
          <div class="p-datatable-table-container" style="overflow: auto;" data-pc-section="tablecontainer">
            <table role="table" class="p-datatable-table" style="min-width: 50rem;" data-pc-section="table">
              <thead class="p-datatable-thead" style="position: sticky;" role="rowgroup" data-pc-section="thead">
                <tr role="row" data-pc-section="headerrow">
                  <th class="p-datatable-header-cell p-datatable-sortable-column" tabindex="0" role="columnheader" aria-sort="none" data-pc-section="headercell" data-pc-name="headercell" pc122="" data-p-sortable-column="true" data-p-resizable-column="false" data-p-sorted="false" data-p-filter-column="false" data-p-reorderable-column="false" first="0">
                    <!---->
                    <div class="p-datatable-column-header-content" data-pc-section="columnheadercontent">
                      <!----><span class="p-datatable-column-title" data-pc-section="columntitle">Name</span><span data-pc-section="sort"><svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg" class="p-icon p-datatable-sort-icon" aria-hidden="true" sorted="false" sortOrder="0" data-pc-section="sorticon"><path d="M5.64515 3.61291C5.47353 3.61291 5.30192 3.54968 5.16644 3.4142L3.38708 1.63484L1.60773 3.4142C1.34579 3.67613 0.912244 3.67613 0.650309 3.4142C0.388374 3.15226 0.388374 2.71871 0.650309 2.45678L2.90837 0.198712C3.17031 -0.0632236 3.60386 -0.0632236 3.86579 0.198712L6.12386 2.45678C6.38579 2.71871 6.38579 3.15226 6.12386 3.4142C5.98837 3.54968 5.81676 3.61291 5.64515 3.61291Z" fill="currentColor"></path><path d="M3.38714 14C3.01681 14 2.70972 13.6929 2.70972 13.3226V0.677419C2.70972 0.307097 3.01681 0 3.38714 0C3.75746 0 4.06456 0.307097 4.06456 0.677419V13.3226C4.06456 13.6929 3.75746 14 3.38714 14Z" fill="currentColor"></path><path d="M10.6129 14C10.4413 14 10.2697 13.9368 10.1342 13.8013L7.87611 11.5432C7.61418 11.2813 7.61418 10.8477 7.87611 10.5858C8.13805 10.3239 8.5716 10.3239 8.83353 10.5858L10.6129 12.3652L12.3922 10.5858C12.6542 10.3239 13.0877 10.3239 13.3497 10.5858C13.6116 10.8477 13.6116 11.2813 13.3497 11.5432L11.0916 13.8013C10.9561 13.9368 10.7845 14 10.6129 14Z" fill="currentColor"></path><path d="M10.6129 14C10.2426 14 9.93552 13.6929 9.93552 13.3226V0.677419C9.93552 0.307097 10.2426 0 10.6129 0C10.9833 0 11.2904 0.307097 11.2904 0.677419V13.3226C11.2904 13.6929 10.9832 14 10.6129 14Z" fill="currentColor"></path></svg></span>
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </th>
                  <th style="min-width: 8rem;" class="p-datatable-header-cell" role="columnheader" data-pc-section="headercell" data-pc-name="headercell" pc124="" data-p-resizable-column="false" data-p-sorted="false" data-p-filter-column="false" data-p-reorderable-column="false" first="0">
                    <!---->
                    <div class="p-datatable-column-header-content" data-pc-section="columnheadercontent">
                      <!----><span class="p-datatable-column-title" data-pc-section="columntitle">Actions</span>
                      <!---->
                      <!---->
                      <!---->
                      <!---->
                    </div>
                  </th>
                </tr>
                <!---->
              </thead>
              <!---->
              <tbody class="p-datatable-tbody" role="rowgroup" data-pc-section="tbody">
                <!---->
                <tr class="p-row-even" tabindex="-1" role="row" data-pc-section="bodyrow" data-p-index="0" data-p-selectable-row="false">
                  <td class="" role="cell" data-pc-section="bodycell" data-pc-name="bodycell" pc127="" data-p-selection-column="false" data-p-cell-editing="false">Test Cohort</td>
                  <td style="min-width: 8rem;" class="" role="cell" data-pc-section="bodycell" data-pc-name="bodycell" pc128="" data-p-selection-column="false" data-p-editable-column="false" data-p-cell-editing="false"><button data-v-2fbb3249="" class="p-button p-component p-button-icon-only p-button-rounded p-button-outlined mr-2" type="button" data-pc-name="button" data-p-disabled="false" pc134="" data-pc-section="root" data-pd-tooltip="true"><span class="p-button-icon pi pi-pencil" data-pc-section="icon"></span><span class="p-button-label" data-pc-section="label">&nbsp;</span>
                      <!---->
                    </button><button data-v-2fbb3249="" class="p-button p-component p-button-icon-only p-button-danger p-button-rounded p-button-outlined mr-2" type="button" data-pc-name="button" data-p-disabled="false" data-p-severity="danger" pc135="" data-pc-section="root" data-pd-tooltip="true"><span class="p-button-icon pi pi-trash" data-pc-section="icon"></span><span class="p-button-label" data-pc-section="label">&nbsp;</span>
                      <!---->
                    </button>
                    <!--v-if-->
                  </td>
                </tr>
                <!---->
                <!---->
              </tbody>
              <!---->
              <!---->
            </table>
          </div>
          <!---->
          <!---->
          <div class="p-datatable-column-resize-indicator" style="display: none;" data-pc-section="columnresizeindicator"></div>
          <!---->
          <!---->
        </div>
      </div>
      <!---->
    </div>
  </transition-stub>
</div>
<!-- Notes dialog -->
<!--teleport start-->
<!--teleport end-->
<!-- Edit item dialog -->
<!--teleport start-->
<!--teleport end-->"
`;
